File: apps/web/src/types/payment.ts
export type PaymentType = "ADVANCE" | "PARTIAL" | "FINAL";
export type PaymentStatus = "PENDING" | "CONFIRMED" | "FAILED";

export interface Payment {
  id: string;
  project_id: string;
  user_id: string;
  type: PaymentType;
  status: PaymentStatus;
  amount: string;
  wompi_ref: string | null;
  wompi_id: string | null;
  confirmed_at: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface PaymentWithCheckout extends Payment {
  checkout_url: string;
}

export const PAYMENT_TYPE_LABELS: Record<PaymentType, string> = {
  ADVANCE: "Anticipo",
  PARTIAL: "Pago parcial",
  FINAL: "Pago final",
};

export const PAYMENT_STATUS_LABELS: Record<PaymentStatus, string> = {
  PENDING: "Pendiente",
  CONFIRMED: "Confirmado",
  FAILED: "Fallido",
};

export const PAYMENT_STATUS_COLORS: Record<PaymentStatus, string> = {
  PENDING:   "bg-yellow-100 text-yellow-800 dark:bg-yellow-900/40 dark:text-yellow-300",
  CONFIRMED: "bg-emerald-100 text-emerald-800 dark:bg-emerald-900/40 dark:text-emerald-300",
  FAILED:    "bg-red-100 text-red-800 dark:bg-red-900/40 dark:text-red-300",
};

File: apps/api/models/payment.py
import enum
from datetime import datetime, timezone
from decimal import Decimal
from sqlalchemy import String, Numeric, DateTime, ForeignKey, func
from sqlalchemy.orm import Mapped, mapped_column

from db.session import Base


class PaymentType(str, enum.Enum):
    ADVANCE = "ADVANCE"
    PARTIAL = "PARTIAL"
    FINAL = "FINAL"


class PaymentStatus(str, enum.Enum):
    PENDING = "PENDING"
    CONFIRMED = "CONFIRMED"
    FAILED = "FAILED"


def _now() -> datetime:
    return datetime.now(timezone.utc).replace(tzinfo=None)


class Payment(Base):
    __tablename__ = "payments"

    id: Mapped[str] = mapped_column(String, primary_key=True)
    project_id: Mapped[str] = mapped_column("project_id", String, ForeignKey("projects.id"), nullable=False)
    user_id: Mapped[str] = mapped_column("user_id", String, ForeignKey("users.id"), nullable=False)
    type: Mapped[str] = mapped_column(String, nullable=False)
    status: Mapped[str] = mapped_column(String, default="PENDING")
    amount: Mapped[Decimal] = mapped_column(Numeric(18, 2), nullable=False)
    wompi_ref: Mapped[str | None] = mapped_column("wompi_ref", String, unique=True, nullable=True)
    wompi_id: Mapped[str | None] = mapped_column("wompi_id", String, nullable=True)
    confirmed_at: Mapped[datetime | None] = mapped_column("confirmed_at", DateTime, nullable=True)
    notes: Mapped[str | None] = mapped_column(String, nullable=True)
    billing_company: Mapped[str | None] = mapped_column("billing_company", String, nullable=True)
    billing_nit: Mapped[str | None] = mapped_column("billing_nit", String, nullable=True)
    billing_email: Mapped[str | None] = mapped_column("billing_email", String, nullable=True)
    billing_phone: Mapped[str | None] = mapped_column("billing_phone", String, nullable=True)
    created_at: Mapped[datetime] = mapped_column("created_at", DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column("updated_at", DateTime, default=_now, onupdate=_now)
